nselib/httpspider.lua

```diff
@@ -938,6 +972,10 @@ Crawler = {
     if ( nil == self.options.doscraping ) then
       self.options.doscraping = stdnse.get_script_args(ln .. ".doscraping")
     end
+    if ( nil == self.options.docookies ) then
+      self.options.docookies = stdnse.get_script_args(ln .. ".docookies")
```

Comment:
Ok, so the library by default won't overwrite cookies. Isn't the opposite of your '.docookies'?  We don't need another argument. If overwriting cookies is allowed ( overwrite_cookie = true), then we do the behavior you implemented in docookies.